lib/checkCriticalPullRequest.js

```diff
@@ -0,0 +1,137 @@
+const {
+  getAllChangedFiles,
+  hasCriticalLabel,
+  CRITICAL_LABEL,
+} = require('./checkPullRequestJob');
+const { teamLeads } = require('../userWhitelist.json');
+const STORAGE_DIR_PREFIX = 'core/storage/';
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem} file
+ */
+const getNewModelsFromFile = (file) => {
+  const newLine = '\n';
+  const changesArray = file.patch.split(newLine);
+  const modelRegex = /(?<addition>\+)(?<classDefinition>class\s)(?<modelName>[a-zA-Z]{2,256})(?<modelSuffix>Model)(?<funDef>\()/;
+
+  const newModelDefinitions = changesArray.filter(change => {
+    const matches = modelRegex.exec(change);
+    return matches !== null;
+  });
+
+  const newModels = newModelDefinitions.map(definition => {
+    const matches = modelRegex.exec(definition);
+    return matches.groups.modelName + matches.groups.modelSuffix;
+  });
+
+  return newModels;
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem} file
+ */
+const addsNewModel = (file) => {
+  const newModels = getNewModelsFromFile(file);
+  return newModels.length > 0;
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} newModelFiles
+ */
+const getModelNameString = (newModelFiles) => {
+  let modelNameString = '';
+  let totalNumberOfModels = 0;
+
+  let modelNameArray = newModelFiles.map(file => {
+    const newModelNames = getNewModelsFromFile(file);
+    totalNumberOfModels += newModelNames.length;
+    return (newModelNames.join(', ')).link(file.blob_url);
+  });
+
+  if (totalNumberOfModels === 1) {
+    modelNameString = ' The name of the model is ' + modelNameArray[0] + '.';
+  } else {
+    modelNameString = ' The models are ' + modelNameArray.join(', ') + '.';
+  }
+
+  return modelNameString;
```

Comment:
Ditto here - let us try to merge functionality as much as possible into a util module so that it can be used for general purpose.

Reply:
Done.